src/coreComponents/physicsSolvers/SolverBase.cpp

```diff
@@ -473,6 +471,8 @@ bool SolverBase::lineSearch( real64 const & time_n,
                              real64 const scaleFactor,
                              real64 & lastResidual )
 {
+  Timer timer( m_timers["line search"] );
```

Comment:
if we want the` line search` to be efficient we should really avoid re-assemblying the matrix.